pygmt/src/inset.py

```diff
@@ -0,0 +1,106 @@
+"""
+inset - Create inset figures.
+"""
+import contextlib
+
+from pygmt.clib import Session
+from pygmt.helpers import build_arg_string, fmt_docstring, kwargs_to_strings, use_alias
+
+
+@fmt_docstring
+@contextlib.contextmanager
+@use_alias(D="position", F="box", M="margin", V="verbose")
+@kwargs_to_strings(D="sequence", M="sequence")
+def inset(self, **kwargs):
+    r"""
+    Create an inset within a figure to plot an additional figure.
+
+    This function create the location, frame, and margins for an inset
+    figure. This function can only be called once the original figure is
+    created. Plotting functions that are called within the context manager
+    are added to the inset figure.
```

Comment:
The original GMT docs at https://docs.generic-mapping-tools.org/dev/inset.html doesn't translate well to the PyGMT world for `inset`, but we can probably make better description here.

1. Change "location" -> "position", and try to describe what an `inset` is without using the word `inset` (as in a dictionary).
2. Delete the line "This function can only be called once the original figure is created". The only way to call inset is using `fig.inset` so there is a figure already by default.
3. Not everyone knows what a 'context manager' is, maybe give an example code snippet. You can make a docstring test that looks like so:

```python
Examples
--------

> with fig.inset(position=...):
> 
```

See e.g. https://github.com/GenericMappingTools/pygmt/blob/55ade6f6b0c5e589a43a59bca8c7d55569939f12/pygmt/datasets/earth_relief.py#L58-L68

Reply:
Made the changes; let me know what you think!

Reply:
Yep, looks much better! Just a few stylistic comments at https://github.com/GenericMappingTools/pygmt/pull/788#discussion_r567631459 to improve it.